contracts/lockdrop/src/contract.rs

```diff
@@ -1513,6 +1664,63 @@ pub fn callback_deposit_liquidity_in_astroport(
         ]))
 }
 
+fn callback_distribute_asset_reward(
+    mut deps: DepsMut,
+    env: Env,
+    previous_balance: Uint128,
+    terraswap_lp_token: Addr,
+    user_address: Addr,
+    lock_duration: u64,
+) -> StdResult<Response> {
+    let reward_balance =
+        astroport::querier::query_balance(&deps.querier, env.contract.address, "uusd".to_string())?;
+    let latest_reward_amount = reward_balance - previous_balance;
```

Comment:
If we did not receive rewards, then latest_reward_amount will be zero. We should add a check for this one, and don't save the state if total_bluna_reward  has not changed.

Reply:
I fixed this case